toolbase.js

```diff
@@ -344,7 +344,34 @@ tools.add({
 	hideInToolbox: true,
 	hideOnSimpleMode: true,
 	tests: () => {
-		tools.expect(source("color")).toBe("color");
+		tools.expect(target("color")).toBe("color");
+	}
+})
+
+function metadata(propertyName) {
+	if (typeof _metadata == 'undefined') return propertyName;
+	return _metadata[propertyName];
```

Comment:
index reading from null array/object produces null pointer in javascript, please not only check for undefined but also null